cloud/pkg/cloudhub/common/helper_test.go

```diff
@@ -0,0 +1,263 @@
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			msg := beehivemodel.NewMessage("")
+			msg.SetResourceOperation(test.resource, "operation")
+			TrimMessage(msg)
+			assert.Equal(t, test.stdResult, msg.GetResource())
+		})
+	}
+}
+
+func TestConstructConnectMessage(t *testing.T) {
+	nodeID := "test-node-id"
+	info := &model.HubInfo{NodeID: nodeID}
+
+	msg := ConstructConnectMessage(info, true)
+	assert.NotNil(t, msg)
+	assert.Equal(t, model.SrcCloudHub, msg.GetSource())
+	assert.Equal(t, model.GpResource, msg.GetGroup())
+	assert.Equal(t, model.NewResource(model.ResNode, nodeID, nil), msg.GetResource())
+
+	var body map[string]interface{}
+	err := json.Unmarshal(msg.GetContent().([]byte), &body)
+	assert.NoError(t, err)
+	assert.Equal(t, model.OpConnect, body["event_type"])
+	assert.Equal(t, nodeID, body["client_id"])
+}
+
+func TestDeepCopy(t *testing.T) {
+	msg := beehivemodel.NewMessage("sample message")
+	msg.FillBody("sample content")
+
+	copy := DeepCopy(msg)
+	assert.NotNil(t, copy)
+	assert.Equal(t, msg.GetID(), copy.GetID())
+	assert.Equal(t, msg.GetContent(), copy.GetContent())
+}
+
+func TestAckMessageKeyFunc(t *testing.T) {
+	cases := []struct {
+		name      string
+		obj       interface{}
+		stdResult string
+		hasError  bool
+	}{
+		{
+			name: "Valid message with GroupResource",
+			obj: &beehivemodel.Message{
+				Header: beehivemodel.MessageHeader{ID: "test-id"},
+				Router: beehivemodel.MessageRoute{Group: edgecon.GroupResource},
+				Content: &v1.ObjectMeta{
+					UID: "test-uid",
+				},
+			},
+			stdResult: "test-uid",
+			hasError:  false,
+		},
+		{
+			name:      "Invalid object type",
+			obj:       "invalid",
+			stdResult: "",
+			hasError:  true,
+		},
+		{
+			name: "Message without GroupResource",
+			obj: &beehivemodel.Message{
+				Header: beehivemodel.MessageHeader{ID: "test-id"},
+				Router: beehivemodel.MessageRoute{Group: "other-group"},
+			},
+			stdResult: "",
+			hasError:  true,
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			result, err := AckMessageKeyFunc(test.obj)
+			if test.hasError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, test.stdResult, result)
+		})
+	}
+}
+
+func TestNoAckMessageKeyFunc(t *testing.T) {
+	cases := []struct {
+		name      string
+		obj       interface{}
+		stdResult string
+		hasError  bool
+	}{
+		{
+			name: "Valid message",
+			obj: &beehivemodel.Message{
+				Header: beehivemodel.MessageHeader{ID: "test-id"},
+			},
+			stdResult: "test-id",
+			hasError:  false,
+		},
+		{
+			name:      "Invalid object type",
+			obj:       "invalid",
+			stdResult: "",
+			hasError:  true,
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			result, err := NoAckMessageKeyFunc(test.obj)
+			if test.hasError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, test.stdResult, result)
+		})
+	}
+}
```

Comment:
run `go fmt` for this file to fix lint

Reply:
I always use `gofmt` and `gci` before pushing to avoid linting issues. I did that again, one empty line added at the end. hopefully this fixes lint.